### DISCUSSION TITLE
[Idea] Add a download fold button as sometimes huge repository’s contain examples standalone apps

### DISCUSSION THREAD

**Discussion by davidbuckleyni** [PrestaShop/PrestaShop#37453] [Ideas]
### The problem

Please add a way to download a whole folder via the gui.  I know this can already be done via cli take for example uno repo here u see there is a sample app Ano u can get it the folder by cli but would prefer a button to be accessible at folder level

if there is sub folders the button should ask do u want to include sub folders 

![IMG_8794](https://github.com/user-attachments/assets/f3f5f309-05ef-4cc0-b8b9-60d627776ab2)

### The solution

Have a button when inside folders show a popup with a check box download sub folders compress as a zip file for the user to use the sample

### Alternatives

To download only a specific directory from a GitHub repository without cloning the entire repo, you can use the following methods:

1.	⁠Sparse Checkout with Git CLI: This method allows you to clone only the desired directory. Here's how:

git init git remote add origin <repository_url> git config core.sparseCheckout true echo "<directory_path>/*" >> .git/info/sparse-checkout git pull origin main

Replace <repository_url> with the URL of the repository and <directory_path> with the path to the directory you want to download. This approach is detailed in a Stack Overflow discussion.

2. Using DownGit: DownGit is a web tool that enables downloading specific directories or files from GitHub. Simply enter the URL of the directory you wish to download, and it will generate a direct download link.

3. Using degit: If you're comfortable with Node.js, degit is a tool that allows you to clone a specific directory from a repository. Install it using npm:

npm install -g degit

Then, use it to clone the desired directory:

degit <repository_url>/<directory_path> <destination_folder>

Replace <repository_url>, <directory_path>, and <destination_folder> accordingly.

These methods should help you retrieve only the specific directory you need without downloading the entire repository.​​

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by matks** [comment]
Hello @davidbuckleyni are you sure you opened this discussion in the right place? this repository is about the prestashop project
